.github/workflows/validation-tests.yml

```diff
@@ -34,78 +119,248 @@ jobs:
   validation-build-mtl:
     runs-on: [Linux, self-hosted, DPDK]
     timeout-minutes: 60
-
+    outputs:
+      pipenv-activate: ${{ steps.pipenv-install.outputs.VIRTUAL_ENV }}
     steps:
-      - name: Harden Runner
+      - name: 'preparation: Harden Runner'
         uses: step-security/harden-runner@91182cccc01eb5e619899d80e4e971d6181294a7 # v2.10.1
         with:
           egress-policy: audit
 
-      - name: Checkout MTL
+      - name: 'preparation: Restore valid repository owner and print env'
+        if: always()
+        run: |
+          sudo chown -R "${USER}" "$(pwd)"
+          env | grep TEST_
+
+      - name: 'preparation: Checkout MTL'
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
           ref: '${{ inputs.branch-to-checkout }}'
 
-      - name: Checkout DPDK
+      - name: 'preparation: Checkout DPDK'
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
           repository: 'DPDK/dpdk'
           ref: 'v${{  env.DPDK_VERSION  }}'
           path: 'dpdk'
 
-      - name: Install the build dependency
+      - name: 'configuration: Install the build dependency'
         run: |
-          sudo apt-get install -y git gcc meson python3 python3-pyelftools pkg-config libnuma-dev libjson-c-dev libpcap-dev libgtest-dev libsdl2-dev libsdl2-ttf-dev libssl-dev
-          sudo apt-get install -y systemtap-sdt-dev pipenv
+          sudo apt update
```

Comment:
i think such long shell commands should be encapsulet in script and kept in repo